dotnet/src/Plugins/Plugins.Web/Bing/BingTextSearch.cs

```diff
@@ -0,0 +1,270 @@
+
+        long? totalCount = searchOptions.IncludeTotalCount ? searchResponse?.WebPages?.TotalEstimatedMatches : null;
+
+        return new KernelSearchResults<object>(this.GetResultsAsWebPageAsync(searchResponse, cancellationToken), totalCount, GetResultsMetadata(searchResponse));
+    }
+
+    #region private
+
+    private readonly ILogger _logger;
+    private readonly HttpClient _httpClient;
+    private readonly string? _apiKey;
+    private readonly Uri? _uri = null;
+    private readonly MapBingWebPageToString _mapToString;
+    private readonly MapBingWebPageToTextSearchResult _mapToTextSearchResult;
+
+    // See https://learn.microsoft.com/en-us/bing/search-apis/bing-web-search/reference/query-parameters
+    private static readonly string[] s_queryParameters = ["answerCount", "cc", "freshness", "mkt", "promote", "responseFilter", "safeSearch", "setLang", "textDecorations", "textFormat"];
+
+    private const string DefaultUri = "https://api.bing.microsoft.com/v7.0/search";
+    private const string SiteQueryParam = "site";
+
+    /// <summary>
+    /// Execute a Bing search query and return the results.
+    /// </summary>
+    /// <param name="query">What to search for.</param>
+    /// <param name="searchOptions">Search options.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    private async Task<BingSearchResponse<BingWebPage>?> ExecuteSearchAsync(string query, TextSearchOptions searchOptions, CancellationToken cancellationToken = default)
+    {
+        using HttpResponseMessage response = await this.SendGetRequestAsync(query, searchOptions, cancellationToken).ConfigureAwait(false);
+
+        this._logger.LogDebug("Response received: {StatusCode}", response.StatusCode);
+
+        string json = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+
+        // Sensitive data, logging as trace, disabled by default
+        this._logger.LogTrace("Response content received: {Data}", json);
+
+        return JsonSerializer.Deserialize<BingSearchResponse<BingWebPage>>(json);
+    }
+
+    /// <summary>
+    /// Sends a GET request to the specified URI.
+    /// </summary>
+    /// <param name="query">The query string.</param>
+    /// <param name="searchOptions">The search options.</param>
+    /// <param name="cancellationToken">A cancellation token to cancel the request.</param>
+    /// <returns>A <see cref="HttpResponseMessage"/> representing the response from the request.</returns>
+    private async Task<HttpResponseMessage> SendGetRequestAsync(string query, TextSearchOptions searchOptions, CancellationToken cancellationToken = default)
+    {
+        var count = searchOptions.Count;
+        var offset = searchOptions.Offset;
+
+        if (count is <= 0 or >= 50)
```

Comment:
I wonder if the `if (count <= 0 or >= 50)` condition is right. Should it not be `if (count <= 0 or > 50)` to allow the API consumers to specify 50 as the page size instead of 49? Is 49 the maximum page size Bing can return?